util/conditions/v1beta2/matcher.go

```diff
@@ -0,0 +1,146 @@
+/*
```

Comment:
Is it possible to import exported code from _test.go files of other packages?

(I think the goal here is that others can use the matcher)

Reply:
If we add _test, it can't be imported from other packages
Also, it is already the case for existing condition utils, so we are not making things worst (but I agree we should consider a better solution sometimes down the path)

Reply:
fine for me